ocean/src/clans.rs

```diff
@@ -1,39 +1,76 @@
+use std::collections::HashMap;
 
 #[derive(Debug)]
 pub struct ClanSystem {
     // TODO: add necessary fields
+    my_map: HashMap<String, Vec<String>>,
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem{ my_map : HashMap::new() }
+    }
+
+    pub fn add_member_to_clan(&mut self, clan_id: &str, crab_name: &str) {
+        if(self.my_map.contains_key(clan_id)) {
+            let list = self.my_map.get_mut(clan_id);
+            list.unwrap().push(crab_name.to_string());
+        } else {
+            let mut list: Vec<String> = Vec::new();
+            list.push(crab_name.to_string());
+            self.my_map.insert(clan_id.to_string(), list);
```

Comment:
Nice job with containing the logic for handling the requirement of "If a clan with this id does not exist, create one" within the same function! For the given specs, this wraps it up well. If requirements expanded (beyond the scope of this assignment) for other use cases of creating a clan, this could be split up into a separate function for modularity